VidLoader/VidLoader/Sources/Utils/CookieOptionsUtils.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  CookieOptionsUtils.swift
+//  VidLoader
+//
+//  Created by Emmanouil Nicolas on 10/05/22.
+//  Copyright © 2022 Petre. All rights reserved.
+//
+
+import AVFoundation
+
+struct CookieOptionsUtils {
+    
+    static func insertCookieOptionsWith(domain: String?, headers: [String: String]?) {
+        guard let domain = domain else { return }
+        
+        if let headers = headers {
+            for key in headers.keys {
+                let cookie: [HTTPCookiePropertyKey : Any] = [
+                    HTTPCookiePropertyKey.domain: domain,
+                    HTTPCookiePropertyKey.path: "/",
+                    HTTPCookiePropertyKey.secure: true,
```

Comment:
This looks project specific if cookie creation is needed only for `AVURLAsset(url: url, options: options)` then you can easily add ne property in `ItemInformation` -> something like:
`let AVURLAssetHTTPHeaderFieldsKey: [Any]?`

and pass to 

`func urlAsset(with mediaURL: URL?, headerFields: [Any]?) {
   -----///-----
     if let headerFields = headerFields {
         options["AVURLAssetHTTPHeaderFieldsKey"] = headerFields
      }
}
`